### PR TITLE
Updates Figures Admin interface + CHANGELOG

### PR DESCRIPTION
**Pull request by johnbaldwin**
## Git log:

* Adds filtering to Figures metrics model admin views
* Adds droplist when number of choices > 3 (Hardcoded in the partial)
* Adds link to user if metrics model has a populated user field
* Updated test coverage to test the admin views "user_link" method
* Updated changelog for upcoming 0.2.0 release
* Tiny bit of code cleanup

### CODE REVIEW

File: tests/views/test_general_course_data_view.py [hunk 1]
```diff
@@ -160,8 +160,6 @@ def test_get_course_in_other_site(self, monkeypatch):
         This test is broken. Needs review of django.contrib.sites handling for
         `get_current_site`
         """
-        def test_site(request):
-            return alpha_site
```
Comment (johnbaldwin) [inline tests/views/test_general_course_data_view.py]: Removed dead code

**Comment by johnbaldwin** [comment]
[Improve admin interface for Figures](https://trello.com/c/O7B5j2fi/761-improve-admin-interface-for-figures)

File: figures/admin.py [hunk 2]
```diff
@@ -1,41 +1,87 @@
...
+    AllValuesFieldListFilter,
+    RelatedOnlyFieldListFilter)
+from django.core.urlresolvers import reverse
+from django.utils.html import format_html
 
 import figures.models
 
 
+class AllValuesDropdownFilter(AllValuesFieldListFilter):
+    """Show droplist when there are more than three filter choices for this field
```
Comment (OmarIthawi) [inline figures/admin.py]:
```suggestion
    """Show dropdown when there are more than three filter choices for this field
```

First time I hear about `droplist`. If it's a synonym to `dropdown`, then I think we probably should use the more common alternative.

File: figures/templates/figures/admin_dropdown_filter.html [hunk 3]
```diff
@@ -0,0 +1,28 @@
+{# Attribution: Original code here: https://github.com/mrts/django-admin-list-filter-dropdown #}
+
+{% load i18n %}
+<script type="text/javascript">
+    var go_from_select = function(opt) { window.location = window.location.pathname + opt };
```
Comment (OmarIthawi) [inline figures/templates/figures/admin_dropdown_filter.html]:
```suggestion
    var figures_go_from_select = function(opt) { window.location = window.location.pathname + opt };
```

Just a precautionary prefix.

File: figures/templates/figures/admin_dropdown_filter.html [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+</script>
+<h3>{% blocktrans with title as filter_title %} By {{ filter_title }} {% endblocktrans %}</h3>
+<ul class="admin-filter-{{ title|cut:' ' }}">
+{# Set the number after slice to the number of choices less one that trigger the selection list #}
+{% if choices|slice:"4:" %}
+    <li>
+    <select class="form-control" style="width: 95%;margin-left: 2%;"
+        onchange="go_from_select(this.options[this.selectedIndex].value)">
```
Comment (OmarIthawi) [inline figures/templates/figures/admin_dropdown_filter.html]: Same.

```suggestion
        onchange="figures_go_from_select(this.options[this.selectedIndex].value)">
```

File: figures/templates/figures/admin_dropdown_filter.html [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    {% for choice in choices %}
+        <option{% if choice.selected %} selected="selected"{% endif %}
+         value="{{ choice.query_string|iriencode }}">{{ choice.display }}</option>
```
Comment (OmarIthawi) [inline figures/templates/figures/admin_dropdown_filter.html]: `iriencode`! never heard about that before! Thanks for adding it.

File: figures/templates/figures/admin_dropdown_filter.html [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+    {% endfor %}
+    </select>
+    </li>
+{% else %}
+    <h3>Default</h3>
+    {% for choice in choices %}
+            <li{% if choice.selected %} class="selected"{% endif %}>
+            <a href="{{ choice.query_string|iriencode }}">{{ choice.display }}</a></li>
+    {% endfor %}
+
```
Comment (OmarIthawi) [inline figures/templates/figures/admin_dropdown_filter.html]: Seems inconsistent.

```suggestion
```

File: figures/templates/figures/admin_dropdown_filter.html [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+    {% for choice in choices %}
+        <option{% if choice.selected %} selected="selected"{% endif %}
+         value="{{ choice.query_string|iriencode }}">{{ choice.display }}</option>
+    {% endfor %}
+    </select>
+    </li>
+{% else %}
+    <h3>Default</h3>
+    {% for choice in choices %}
+            <li{% if choice.selected %} class="selected"{% endif %}>
```
Comment (OmarIthawi) [inline figures/templates/figures/admin_dropdown_filter.html]:
```suggestion
            <li {% if choice.selected %}class="selected"{% endif %}>
```
Spaces are both cheap and beautiful.

File: figures/templates/figures/admin_dropdown_filter.html [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+</script>
+<h3>{% blocktrans with title as filter_title %} By {{ filter_title }} {% endblocktrans %}</h3>
+<ul class="admin-filter-{{ title|cut:' ' }}">
+{# Set the number after slice to the number of choices less one that trigger the selection list #}
+{% if choices|slice:"4:" %}
+    <li>
+    <select class="form-control" style="width: 95%;margin-left: 2%;"
+        onchange="go_from_select(this.options[this.selectedIndex].value)">
+    {% for choice in choices %}
+        <option{% if choice.selected %} selected="selected"{% endif %}
```
Comment (OmarIthawi) [inline figures/templates/figures/admin_dropdown_filter.html]:
```suggestion
        <option {% if choice.selected %}selected="selected"{% endif %}
```

File: figures/templates/figures/admin_dropdown_filter.html [hunk 9]
```diff
@@ -0,0 +1,28 @@
+{# Attribution: Original code here: https://github.com/mrts/django-admin-list-filter-dropdown #}
+
+{% load i18n %}
```
Comment (OmarIthawi) [inline figures/templates/figures/admin_dropdown_filter.html]:
```suggestion
```

File: figures/templates/figures/admin_dropdown_filter.html [hunk 10]
```diff
@@ -0,0 +1,28 @@
+{# Attribution: Original code here: https://github.com/mrts/django-admin-list-filter-dropdown #}
```
Comment (OmarIthawi) [inline figures/templates/figures/admin_dropdown_filter.html]: It's a lot safer to use `load` tags before comments. Both Python `import` and Django `load` are "big deal" and usually the order matters, esp. when using something like `from __future__ import unicode_literals`.

Therefore, I think it's just proper hygiene to use `load` first.

```suggestion
{% load i18n %}

{# Attribution: Original code here: https://github.com/mrts/django-admin-list-filter-dropdown #}
```

File: figures/templates/figures/admin_dropdown_filter.html (hunk 3 above)
Comment (johnbaldwin) [reply figures/templates/figures/admin_dropdown_filter.html]: Good call, @OmarIthawi !

File: figures/templates/figures/admin_dropdown_filter.html (hunk 5 above)
Comment (johnbaldwin) [reply figures/templates/figures/admin_dropdown_filter.html]: @OmarIthawi It's in the Django source too: https://github.com/django/django/blob/master/django/contrib/admin/templates/admin/filter.html#L6

**Comment by johnbaldwin** [comment]
@OmarIthawi updated per comments, would you please give it a quick check?

Thanks!!!

File: figures/templates/figures/admin_dropdown_filter.html (hunk 5 above)
Comment (OmarIthawi) [reply figures/templates/figures/admin_dropdown_filter.html]: Now I know, I just didn't know it existed :slightly_smiling_face:

**Pull request merged**